buildSrc/src/main/java/com/google/firebase/gradle/plugins/UpdatePinnedDependenciesTask.kt

```diff
@@ -60,29 +60,61 @@ abstract class UpdatePinnedDependenciesTask : DefaultTask() {
 
   @TaskAction
   fun build() {
```

Comment:
As it stands right now, without context it takes a bit to read through the function to fully understand what's doing. 

- `build` as the function name doesn't match what the action does
- `needsChanging` is not a noun, so when reading through the code is not clear what's stored in the val
- `lines`, `newlines` , and `validateNewLines`. While yes, at the end of the day what will end up happening is lines in a file changing, we can keep this function focus on the "what" should happen(update dependencies), rather that on "how" it should happen (change lines in a file). 

Question:

- Do we need to throw if there's nothing to change? Why is an early return not enough?

Reply:
Could we use `onlyIf` for this scenario?

Reply:
Yes and no.

Yes, we could technically do it- but we shouldn't be able to. The only reason we **are** able to is because of the way our publishing plugin is structured. Additionally, it would require us to provide the project level dependencies as inputs- and would require the task that registers this (the post release task) to know about the existence of `DEPENDENCIES_TO_IGNORE` .

Reply:
curious on why we shouldn't be able to. It's a task that implements the check, and the check is ran at execution time.

Reply:
On second thought, we could actually make it work if we wrapped the `librariesToUpdate` or w/e in a `Provider`. The initial problem was that while `onlyIf` evaluates at execution time- the data would need to be used outside of the `onlyIf`- as a task input. And that would need to be configured at configuration time. But that's a non-issue if we wrap it in a `Provider`.

My only concern here now would be having `PostReleasePlugin` be aware of `DEPENDENCIES_TO_IGNORE`, although tbf It's not *that* big of a deal- since it *is* an implementation detail.

I'm content either way, wdywd?

Reply:
Thanks for the explanation. It's OK to go with our current approach then.